aws-datastore/src/main/java/com/amplifyframework/datastore/storage/sqlite/SQLiteStorageAdapter.java

```diff
@@ -667,30 +715,44 @@ private void bindPreCompiledInsertStatementWithJavaFields(
 
     // Extract the values of the fields of a model and bind the values to the SQLiteStatement
     // and execute the statement.
-    private <T extends Model> void saveModel(
+    // Return true if a model is successfully saved, false otherwise.
+    private <T extends Model> boolean saveModel(
             @NonNull T model,
             @NonNull ModelSchema modelSchema,
             @NonNull SqlCommand sqlCommand,
-            ModelConflictStrategy modelConflictStrategy) throws IllegalAccessException, DataStoreException {
+            ModelConflictStrategy modelConflictStrategy
+    ) throws IllegalAccessException, DataStoreException {
         Objects.requireNonNull(model);
         Objects.requireNonNull(modelSchema);
         Objects.requireNonNull(sqlCommand);
         Objects.requireNonNull(sqlCommand.getCompiledSqlStatement());
 
         LOG.debug("Writing data to table for: " + model.toString());
+        final boolean writeSuccess;
 
         // SQLiteStatement object that represents the pre-compiled/prepared SQLite statements
         // are not thread-safe. Adding a synchronization barrier to access it.
         synchronized (sqlCommand.getCompiledSqlStatement()) {
             final SQLiteStatement compiledSqlStatement = sqlCommand.getCompiledSqlStatement();
             compiledSqlStatement.clearBindings();
-            bindPreparedSQLStatementWithValues(model, sqlCommand);
+
+            // fieldValues is a list of field values that are
+            // applicable when binding to insert statement.
+            final List<Object> fieldValues = new ArrayList<>(extractFieldValuesFromModel(model));
+            if (sqlCommand.hasSelectionArgs()) {
+                // These are additional values to bind in the case of updating SQLite table.
+                fieldValues.addAll(sqlCommand.getSelectionArgs());
```

Comment:
Is there an order of insertion be maintained?

Reply:
Yes. the indexes should match.